### ISSUE TITLE
query params go before the hash

### ISSUE THREAD

**Issue by jiangts** [funcool/bide#21]
When using the hash parameter setting, bide creates URLs that position the `search` (the `?...` fragment) AFTER the hash fragment. In fact, it should be the other way around (see below).

https://superuser.com/questions/498617/does-an-anchor-tag-come-before-the-query-string-or-after

**Comment by rrrnld** [comment]
I would disagree; this is actually very useful behavior as it allows me to pass arguments in a more flexible way; the reload behavior can be controlled by me and I don't have to know all params a route will take in advance. You can still use the "real" query parameters, no?

**Comment by jiangts** [comment]
so I'm using the router to define both html5 history routes on the client side, as well as to define API endpoints I want to hit on the server.

When the query params are in the wrong place, my server's query parser can't find the params so I can't send params in my GET requests.
